Arrangement_on_surface_2/include/CGAL/Arr_polycurve_basic_traits_2.h

```diff
@@ -952,48 +971,33 @@ class Arr_polycurve_basic_traits_2 {
       std::size_t j = 0;
       std::size_t n1 = cv1.number_of_subcurves();
       std::size_t n2 = cv2.number_of_subcurves();
-      Comparison_result is_cv1_left_to_right = comp_endpt(cv1[0]);
-      Comparison_result is_cv2_left_to_right = comp_endpt(cv2[0]);
+      Comparison_result is_cv1_left_to_right = cmp_endpt(cv1[0]);
+      Comparison_result is_cv2_left_to_right = cmp_endpt(cv2[0]);
 
       while ((i < (n1-1)) || (j < (n2-1))) {
-        Point_2 point1, point2;
-        std::size_t cv1_seg_ind, cv2_seg_ind;
-        if (SMALLER == is_cv1_left_to_right) {
-          cv1_seg_ind = i;
-          point1 = max_vertex(cv1[cv1_seg_ind]);
-        }
-        else {
-          cv1_seg_ind = n1 - 1 - i;
-          point1 = max_vertex(cv1[cv1_seg_ind]);
-        }
-        if (SMALLER == is_cv2_left_to_right) {
-          cv2_seg_ind = j;
-          point2 = max_vertex(cv2[cv2_seg_ind]);
-        }
-        else {
-          cv2_seg_ind = n2 - 1 - j;
-          point2 = max_vertex(cv2[cv2_seg_ind]);
-        }
-
+        std::size_t cv1_seg_ind = (SMALLER == is_cv1_left_to_right) ?
+          i : cv1_seg_ind = n1 - 1 - i;
```

Comment:
```suggestion
          i :  n1 - 1 - i;
```
Spotted by this warining in [ I-c-348](https://cgal.geometryfactory.com/CGAL/testsuite/CGAL-6.0.1-Ic-348/Surface_sweep_2/TestReport_gimeno_Ubuntu-Latest-NO_DEPRECATED_CODE.gz)